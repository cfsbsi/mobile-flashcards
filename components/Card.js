import React, {Component} from 'react'
import {View} from 'react-native'
import styled from "styled-components/native/index"

class Card extends Component {

    constructor(props) {
        super(props);
        this.state = {index: 0, screen: 'Answer', disabledButtons: true};
    }

    showAnswer = () => {
        this.setState({screen: 'Question', disabledButtons: false})
    }

    showQuestion = () => {
        this.setState({screen: 'Answer', disabledButtons: true})
    }

    render() {

        const {deck} = this.props.navigation.state.params;
        const {index, screen} = this.state;

        return (
            <MainView>
                <PageMaker>{this.state.index + 1}/{deck.questions.length}</PageMaker>
            <ScreenView>
                <View>
                    {this.state.disabledButtons?
                        (<QuestionAndAnswer>{deck.questions[index].question}</QuestionAndAnswer>):
                        (<QuestionAndAnswer>{deck.questions[index].answer}</QuestionAndAnswer>)
                    }
                    <ViewAnswer onPress={() => this.state.disabledButtons?this.showAnswer():this.showQuestion()}>{screen}</ViewAnswer>
                </View>

                <View>
                    {this.state.disabledButtons?
                        (
                            <GreenDisabledButton disabled={true}><TextButton>Correct</TextButton></GreenDisabledButton>
                        ):
                        (
                            <GreenButton><TextButton>Correct</TextButton></GreenButton>
                        )
                    }

                    {this.state.disabledButtons?
                        (
                            <RedDisabledButton disabled={true}><TextButton>Incorrect</TextButton></RedDisabledButton>
                        ):
                        (
                            <RedButton><TextButton>Incorrect</TextButton></RedButton>
                        )
                    }
                </View>
            </ScreenView>
            </MainView>
        )
    }
}

const MainView = styled.View`
    flex: 1;
`

const ScreenView = styled.View`
    align-items: center;
    justify-content: space-around;
    flex: 1;
`

const PageMaker = styled.Text`
    align-self: flex-start;
    font-size: 17;
    margin-top: 2;
    margin-left: 4;
`


const QuestionAndAnswer = styled.Text`
    align-self: center;
    justify-content: center;
    font-size: 50;
    text-align: center;
`

const ViewAnswer = styled.Text`
    color: red;
    font-weight: bold;
    font-size: 25;
    text-align: center;
`

const GreenButton = styled.TouchableOpacity`
    background-color: green;
    border-radius: 7;
    width: 200;
    padding: 10px 15px;
`

const GreenDisabledButton = styled.TouchableOpacity`
    background-color: green;
    border-radius: 7;
    width: 200;
    padding: 10px 15px;
    opacity: 0.4;
`

const RedButton = styled.TouchableOpacity`
    margin-top: 10;
    background-color: red;
    border-radius: 7;
    width: 200;
    padding: 10px 15px;
`

const RedDisabledButton = styled.TouchableOpacity`
    margin-top: 10;
    background-color: red;
    border-radius: 7;
    width: 200;
    padding: 10px 15px;
    opacity: 0.4
`


const TextButton = styled.Text`
    font-size: 25;
    text-align: center;
    color: white;
`

export default Card;