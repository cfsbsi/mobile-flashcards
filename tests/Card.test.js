import React from 'react';
import Card from '../components/Card';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const state = {
        state: {
            params: {
                deck: {
                    title: 'React',
                    questions: [
                        {
                            question: 'What is React?',
                            answer: 'A library for managing user interfaces'
                        },
                        {
                            question: 'Where do you make Ajax requests in React?',
                            answer: 'The componentDidMount lifecycle event'
                        }
                    ]
                }
            }
        }
    };

    const tree = renderer.create(<Card navigation={state}/>).toJSON();

    expect(tree).toMatchSnapshot();
});