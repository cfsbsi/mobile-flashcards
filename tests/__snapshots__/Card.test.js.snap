// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "fontSize": 17,
          "marginLeft": 4,
          "marginTop": 2,
        },
        undefined,
      ]
    }
  >
    1
    /
    2
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-around",
        },
        undefined,
      ]
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontSize": 50,
              "justifyContent": "center",
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        What is React?
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "red",
              "fontSize": 25,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Answer
      </Text>
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "green",
            "borderRadius": 7,
            "opacity": 0.4,
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
            "width": 200,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 25,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Correct
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 7,
            "marginTop": 10,
            "opacity": 0.4,
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
            "width": 200,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 25,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Incorrect
        </Text>
      </View>
    </View>
  </View>
</View>
`;
